### PR TITLE
Update to the latest `config`

### PR DESCRIPTION
**Pull request by yevhenii-nadtochii**
This PR updates the repository to use the latest `config`.

Mostly, this is done in order to pull in the following changes:

1. [Make published artifacts full-fledged #335](https://github.com/SpineEventEngine/config/pull/335).
2. [Bump dependencies #344](https://github.com/SpineEventEngine/config/pull/344).

Applying of those changes led to the following:

1. `testJar` artifact has disappeared from the list of artifacts published by the modules. This artifact is [optional](https://github.com/SpineEventEngine/config/pull/340) now and not published by default.
2. `protoJar` artifact has also disappeared in modules that don't have any proto definitions.
3. Deleted empty build files.

The library version is bumped to `2.0.0-SNAPSHOT.91`.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -26,7 +26,7 @@ all modules and does not describe the project structure per-subproject.
   <dependency>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:26]: @sonatype-lift ignore

File: pom.xml [hunk 2]
```diff
@@ -26,7 +26,7 @@ all modules and does not describe the project structure per-subproject.
...
     <groupId>com.google.guava</groupId>
     <artifactId>guava</artifactId>
-    <version>31.0.1-jre</version>
+    <version>31.1-jre</version>
     <scope>compile</scope>
   </dependency>
   <dependency>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:32]: @sonatype-lift ignore

File: pom.xml [hunk 3]
```diff
@@ -217,12 +217,12 @@ all modules and does not describe the project structure per-subproject.
...
     <groupId>com.google.protobuf</groupId>
     <artifactId>protoc</artifactId>
-    <version>3.19.3</version>
+    <version>3.19.4</version>
   </dependency>
   <dependency>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:222]: @sonatype-lift ignore

File: pom.xml [hunk 4]
```diff
@@ -242,12 +242,12 @@ all modules and does not describe the project structure per-subproject.
...
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:242]: @sonatype-lift ignore

**Comment by yevhenii-nadtochii** [comment]
@armiol Please take a look.

**Review by armiol — CHANGES_REQUESTED** [review]
As discussed, please ensure that deleting the empty build files is absolutely safe, in terms of using `tests` root project in IDEA and in standalone builds. 

Also, please see my comment.

**Pull request merged**